frontend/public/components/namespace.jsx

```diff
@@ -124,149 +129,377 @@ const fetchNamespaceMetrics = () => {
+      columnManagementID={NamespacesColumnManagementID}
+      aria-label="Namespaces"
+      Header={NamespacesTableHeader}
+      Row={NamespacesRow}
+      virtualize
+    />
+  );
+});
+
+export const NamespacesPage = (props) => {
+  let selectedColumns = new Set(
+    useSelector(({ UI }) => UI.getIn(['columnManagement', NamespacesColumnManagementID])),
+  );
+  if (_.isEmpty(selectedColumns)) {
+    selectedColumns = getNamespacesSelectedColumns();
+  }
+  return (
+    <ListPage
+      {...props}
+      ListComponent={NamespacesList}
+      canCreate={true}
+      createHandler={() => createNamespaceModal({ blocking: true })}
+      columnLayout={{
+        columns: NamespacesTableHeader().map((column) =>
+          _.pick(column, ['title', 'additional', 'id']),
+        ),
+        id: NamespacesColumnManagementID,
+        selectedColumns,
+        type: 'Namespaces',
+      }}
+    />
+  );
+};
+
 export const projectMenuActions = [Kebab.factory.Edit, deleteModal];
 
-const projectColumnClasses = [
-  '', // name
-  classNames('pf-m-hidden', 'pf-m-visible-on-sm'), // display name
-  '', // status
-  classNames('pf-m-hidden', 'pf-m-visible-on-lg'), // requester
-  classNames('pf-m-hidden', 'pf-m-visible-on-xl'), // memory
-  classNames('pf-m-hidden', 'pf-m-visible-on-xl'), // cpu
-  classNames('pf-m-hidden', 'pf-m-visible-on-2xl'), // created
-  Kebab.columnClass,
-];
+const projectColumnManagementID = referenceForModel(ProjectModel);
 
 const projectTableHeader = ({ showMetrics, showActions }) => {
-  return [
-    {
-      title: 'Name',
-      sortField: 'metadata.name',
-      transforms: [sortable],
-      props: { className: projectColumnClasses[0] },
-    },
-    {
-      title: 'Display Name',
-      sortField: 'metadata.annotations["openshift.io/display-name"]',
-      transforms: [sortable],
-      props: { className: projectColumnClasses[1] },
-    },
-    {
-      title: 'Status',
-      sortField: 'status.phase',
-      transforms: [sortable],
-      props: { className: projectColumnClasses[2] },
-    },
-    {
-      title: 'Requester',
-      sortField: "metadata.annotations.['openshift.io/requester']",
-      transforms: [sortable],
-      props: { className: projectColumnClasses[3] },
-    },
-    ...(showMetrics
-      ? [
-          {
-            title: 'Memory',
-            sortFunc: 'namespaceMemory',
-            transforms: [sortable],
-            props: { className: projectColumnClasses[4] },
-          },
-          {
-            title: 'CPU',
-            sortFunc: 'namespaceCPU',
-            transforms: [sortable],
-            props: { className: projectColumnClasses[5] },
-          },
-        ]
-      : []),
-    {
-      title: 'Created',
-      sortField: 'metadata.creationTimestamp',
-      transforms: [sortable],
-      props: { className: projectColumnClasses[6] },
-    },
-    ...(showActions ? [{ title: '', props: { className: projectColumnClasses[7] } }] : []),
-  ];
+  return () => {
```

Comment:
Does this need to return a function instead of just returning the array?

Reply:
It is returning a function. This part is inside `projectTableHeader` and returns a function, inside the function it returns the array.

Reply:
@DaoDaoNoCode I'm asking why we can't just return an array instead :)

Reply:
In other words, why does this need to return a function? (Sorry I phrased it poorly in my original comment.)

Reply:
Sorry for my misunderstanding. After reviewing my code, I think there is no need to return a function here. Issue fixed! Thanks.